src/Controller/ListProductsAction.php

```diff
@@ -14,23 +19,49 @@
 final class ListProductsAction
 {
     public function __construct(
-        private TwigEnvironment $twig,
-        private GuessCurrentLocaleQuery $guessCurrentLocaleQuery,
-        private FetchProductsQuery $fetchProductsQuery,
+        private readonly TwigEnvironment $twig,
+        private readonly CatalogIdStorageInterface $catalogIdStorage,
+        private readonly PimCatalogApiClient $catalogApiClient,
+        private readonly GuessCurrentLocaleQuery $guessCurrentLocaleQuery,
+        private readonly FetchProductsQuery $fetchProductsQuery,
     ) {
     }
 
     #[Route('/products', name: 'products', methods: ['GET'])]
     public function __invoke(Request $request): Response
     {
+        $catalog = $this->getDefaultCatalog();
         $locale = $this->guessCurrentLocaleQuery->guess();
-        $products = $this->fetchProductsQuery->fetch($locale);
+
+        if ($catalog->enabled) {
+            $productsIdentifiers = $this->catalogApiClient->getProductIdentifiers($catalog->id, 10);
+            $products = $this->fetchProductsQuery->fetch($locale, $productsIdentifiers);
+        } else {
+            $products = [];
+        }
 
         return new Response(
             $this->twig->render('products.html.twig', [
                 'locale' => $locale,
                 'products' => $products,
+                'catalog' => $catalog,
             ])
         );
     }
+
+    private function getDefaultCatalog(): Catalog
+    {
+        $catalogId = $this->catalogIdStorage->getCatalogId();
+        if (null === $catalogId) {
+            throw new CatalogNotFoundException('Catalog Id should exist');
```

Comment:
```suggestion
            throw new CatalogNotFoundException();
```
useless, the exception is already explicit.
But that's not a blocker, you can merge without fixing it.